### PR TITLE
58 ocr jdt

### PR DESCRIPTION
**Pull request by JulienDiTria**
add text detection on video feed by the camera

### CODE REVIEW

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        println("after done")
+    }
+
+    @Test
+    fun test_0_grantPermissions() {
+        assert(true)
+    }
+
+    @Test
+    fun test_1_defaultTextValues() {
```
Comment (Bjolidon) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragmentTest.kt:46]: This test seems to run indefinitely on my machine despite the time outs, maybe you should put a global timeout on the test

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerViewModelTester.kt
Comment (Bjolidon) [inline app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerViewModelTester.kt:1]: Is it necessary to create a empty class for that? Couldn't you directly use the ScannerViewModel class?

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt [hunk 2]
```diff
@@ -20,41 +34,266 @@ class ScannerFragment : Fragment() {
...
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        val baseContext = this.context ?: return false
+        ContextCompat.checkSelfPermission(baseContext, it) == PackageManager.PERMISSION_GRANTED
+    }
+
+    /**
+     * Check whether permissions have been granted after the request,
+     * replaces the onRequestPermissionsResult method from the activity
+     */
+    fun requestPermissionsResult(requestCode: Int, perm: Array<String>, results: IntArray) {
```
Comment (Bjolidon) [inline app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt:101]: "perm" and "results" parameters seems to be useless in this function

**Review by Bjolidon — COMMENTED** [review]
As I don't have an Android phone, I unfortunately can't test the text detection on my side. As I don't have an android phone, I unfortunately can't test the feature on my side. Does everything work correctly?

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragmentTest.kt (hunk 1 above)
Comment (JulienDiTria) [reply app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragmentTest.kt:46]: good idea, i put it

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerViewModelTester.kt
Comment (JulienDiTria) [reply app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerViewModelTester.kt:1]: yes likely, i had added some things in it, but i can remove it indeed

File: app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt (hunk 2 above)
Comment (JulienDiTria) [reply app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt:101]: yes, they are not used here, but they match the original prototype for handling the permissionsResults

Comment (Bjolidon) [reply app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt:101]: Ok, then I guess we can let it that way

**Comment by SolalPirelli** [comment]
What's up with the branching here? It might be easier to make this PR target your other PR since it seems to be based on it

**Pull request closed without merging**